lib/inspector.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const connect = process.binding('inspector').connect;
+const EventEmitter = require('events');
+const util = require('util');
+
+if (!connect)
+  throw new Error('Inspector is not available');
+
+const connectionSymbol = Symbol('connectionProperty');
+const messageCallbacksSymbol = Symbol('messageCallbacks');
+const nextIdSymbol = Symbol('nextId');
+const onMessageSymbol = Symbol('onMessage');
+
+class Session extends EventEmitter {
+  constructor() {
+    super();
+    this[connectionSymbol] = null;
+    this[nextIdSymbol] = 1;
+    this[messageCallbacksSymbol] = new Map();
+  }
+
+  connect() {
+    if (this[connectionSymbol])
+      throw new Error('Already connected');
+    this[connectionSymbol] =
+        connect((message) => this[onMessageSymbol](message));
+  }
+
+  [onMessageSymbol](message) {
+    const parsed = JSON.parse(message);
+    if (parsed['id']) {
+      const callback = this[messageCallbacksSymbol].get(parsed['id']);
+      this[messageCallbacksSymbol].delete(parsed['id']);
+      if (callback)
+        callback(parsed['error'] || null, parsed['result'] || null);
+    } else {
+      this.emit(parsed['method'], parsed);
+      this.emit('inspectorNotification', parsed);
+    }
+  }
+
+  post(method, params, callback) {
+    if (typeof method !== 'string')
+      throw new TypeError(
+          '"method" must be a string, got ' + typeof method + ' instead');
+    if (params && util.isFunction(params)) {
```

Comment:
This should be `!callback && typeof params === 'function'` (or `util.isFunction(params)`), as I believe `session.post(method, () => {}, () => {})` should throw an error.